oneview_redfish_toolkit/blueprints/computer_system.py

```diff
@@ -103,6 +103,81 @@ def get_computer_system(uuid):
         return abort(status.HTTP_500_INTERNAL_SERVER_ERROR)
 
 
+@computer_system.route("/redfish/v1/Systems/<uuid>/"
+                       "Actions/ComputerSystem.Reset", methods=["POST"])
+def change_power_state(uuid):
+    """Change the Oneview power state for a specific Server hardware.
+
+        Return ResetType Computer System redfish JSON for a
+        given server hardware UUID.
+        Logs exception of any error and return abort.
+
+        Returns:
+            JSON: Redfish JSON with ComputerSystem ResetType.
+
+        Exceptions:
+            HPOneViewException: When some OneView resource was not found.
+            return abort(404)
+
+            OneViewRedfishError: When occur a power state mapping error.
+            return abort(400)
+
+            Exception: Unexpected error.
+            return abort(500)
+    """
+
+    reset_type = request.form["ResetType"]
+
+    try:
+        # Recover OV connection
+        ov_client = util.get_oneview_client()
+
+        # Gets ServerHardware for given UUID
+        sh = ov_client.server_hardware.get(uuid)
+
+        # Gets the ServerHardwareType of the given server hardware
+        sht = ov_client.server_hardware_types.get(sh['serverHardwareTypeUri'])
+
+        # Build Computer System object and validates it
+        cs = ComputerSystem(sh, sht)
+
+        oneview_power_configuration = \
+            cs.get_oneview_power_configuration(reset_type)
+
+        # Changes the ServerHardware power state
+        ov_client.server_hardware.update_power_state(
```

Comment:
Do we have a test for a failure in this? What happened to those cases of resetting a powered off node...

Reply:
It's necessary to write a test for this case? If yes, I can do.